plugins/operators/model_evaluation/__init__.py

```diff
@@ -671,15 +688,15 @@ def get_scenarios_picker_type(self, ctx, field_name):
             return ShowOptionsMethod.CODE, CustomCodeViewReason.FLOAT_TYPE
         if isinstance(field, fof.BooleanField):
             # example counts {True: 2, None: 135888}
-            counts = ctx.dataset.count_values(field_name)
+            counts = dataset_or_view.count_values(field_name)
             return ShowOptionsMethod.CHECKBOX, {
                 "true": counts.get(True, 0),
                 "false": counts.get(False, 0),
                 "none": counts.get(None, 0),
             }
 
         # Retrieve distinct values (may be slow for large datasets)
-        distinct_values = ctx.dataset.distinct(field_name)
+        distinct_values = dataset_or_view.distinct(field_name)
         distinct_count = len(distinct_values)
```

Comment:
Nit: does it worth considering caching or lazy loading here?

Reply:
Yah, good idea. It may be cached but I'll check it out and follow-up.